File: dist_clean.py
#!/usr/bin/python3
import shutil
import os

def remove_liburing():
    path = os.getcwd()
    liburing_src_path = path + "/thrd-libs/liburing"
    install_path = path + "/deps/liburing"

    if os.path.exists(install_path):
        shutil.rmtree(install_path)

    if os.path.exists(liburing_src_path):
        shutil.rmtree(liburing_src_path)

if __name__ == "__main__":
    os.system("xmake clean")
    remove_liburing()

File: scripts/tasks/dist_clean.lua
function main(...)
    import("core.base.task")
    task.run("clean")

    if os.exists("build") then
        os.rmdir("build")
        os.mkdir("build")
    end

    path = os.curdir()
    liburing_src_path = path .. "/thrd-libs/liburing"
    install_path = path .. "/deps/liburing"

    os.tryrm(install_path)

    log_install_path = path .. "/deps/easyloggingpp"
    os.tryrm(log_install_path)

    os.exec("git submodule deinit --force --all") 
end
